pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -159,22 +152,21 @@ func getMonitoringConsoleDeployment(cr splcommon.MetaObject, spec *enterprisev1.
 		},
 	}
 
-	env := []corev1.EnvVar{
-		{Name: "SPLUNK_CONFIGMAP_HASH",
-			Value: configMapHash,
-		},
-	}
+	env := []corev1.EnvVar{}
+
 	// append labels and annotations from parent
-	splcommon.AppendParentMeta(deploymentMC.Spec.Template.GetObjectMeta(), cr.GetObjectMeta())
+	splcommon.AppendParentMeta(statefulSet.Spec.Template.GetObjectMeta(), cr.GetObjectMeta())
 
 	// update statefulset's pod template with common splunk pod config
-	updateSplunkPodTemplateWithConfig(&deploymentMC.Spec.Template, cr, spec, instanceType, env, secretName)
+	updateSplunkPodTemplateWithConfig(client, &statefulSet.Spec.Template, cr, spec, instanceType, env, secretName)
+
+	statefulSet.SetOwnerReferences(append(statefulSet.GetOwnerReferences(), splcommon.AsOwner(cr, true)))
```

Comment:
Taking care of it in the latest commit

Reply:
Setting it to false in case of multiple owners, on the deletion of any owner the statefulset get deleted and recreated, which we are trying to avoid here